### ISSUE TITLE
False-positive `unsupported-operator` when adding strings to non-strings

### ISSUE THREAD

**Issue by rodya-mirov** [astral-sh/ty#2020]
### Summary

Consider the following output:

```
error[unsupported-operator]: Unsupported `+` operation
  --> [my_files].py:65:24
   |
63 |                 out2 = item.split(n)[1]
64 |             except IndexError:
65 |                 out2 = "_" + counter + ".txt"
   |                        ---^^^-------
   |                        |     |
   |                        |     Has type `Literal[1]`
   |                        Has type `Literal["_"]`
66 |             output = out_path + "/" + out1 + out2
67 |             my_file = Path(output)
   |
info: rule `unsupported-operator` is enabled by default
```

Obviously you can add a string and an int, but it's barfing here.

Still, I'm excited for what this going to do when it's ready. Congrats on the beta release!

**Comment by AlexWaygood** [comment]
Hey, thanks for the report! But unfortunately this

> Obviously you can add a string and an int

is not accurate, I don't think 😄 here's what happens at runtime:

```pycon
>>> "_" + 1
Traceback (most recent call last):
  File "<python-input-0>", line 1, in <module>
    "_" + 1
    ~~~~^~~
TypeError: can only concatenate str (not "int") to str
```

**Comment by rodya-mirov** [comment]
Well I'll be damned. It works in Java lol

Thanks for finding a bug in some weird corner of my project! Love you guys!

**Comment by AlexWaygood** [comment]
Haha you're very welcome! Thanks for using ty!
